planet/cli/cli.py

```diff
@@ -45,6 +47,20 @@ def main(ctx, verbosity, quiet):
     ctx.ensure_object(dict)
     ctx.obj['QUIET'] = quiet
 
+    # Check authentication sources and store the highest priority in ctx.
+    # 1. Environment.
+    try:
+        auth_env = Auth.from_env()
+    except PlanetError:
+        auth_env = None
+    # 2. Secret file.
+    try:
+        auth_file = Auth.from_file()
+    except PlanetError:
+        auth_file = None
+
+    ctx.obj['AUTH'] = auth_env or auth_file
```

Comment:
Now the root "planet" command takes charge of auth. Previously, we deferred until calling the Session constructor.

Reply:
Warnings might be added here, but I think that could also be done in another PR.

Reply:
Having `planet` take charge of auth is a great idea! To minimize code in the CLI wrapper and make it easier to maintain and understand default auth behavior, what about adding the logic for getting auth to the `auth` module, as something like `Auth.default() -> AuthType` then using that function here and also in the `Session` class.